third_party/accessibility/ax/platform/ax_platform_node_mac_unittest.mm

```diff
@@ -46,4 +46,25 @@
   EXPECT_TRUE(native_root != nullptr);
 }
 
+// Test that [AXPlatformNodeCocoa accessbilityRangeForPosition:] doesn't crash.
+// https://github.com/flutter/flutter/issues/102416
+TEST_F(AXPlatformNodeMacTest, AccessibilityRangeForPositionDoesntCrash) {
+  AXNodeData root;
+  root.id = 1;
+  root.relative_bounds.bounds = gfx::RectF(0, 0, 40, 40);
+
+  Init(root);
+  AXNode* root_node = GetRootAsAXNode();
+  ASSERT_TRUE(root_node != nullptr);
+
+  AXPlatformNode* platform_node = AXPlatformNodeFromNode(root_node);
+  ASSERT_TRUE(platform_node != nullptr);
+
+  NSPoint point = NSMakePoint(0, 0);
+  AXPlatformNodeCocoa* native_root = platform_node->GetNativeViewAccessible();
+  ASSERT_TRUE(native_root != nullptr);
+
+  [native_root accessibilityRangeForPosition:(NSPoint)point];
```

Comment:
A nice improvement to this test would be to actually verify that a proper range is returned.

On the node above you could add:
```cc
std::string value = "Hello";
root.AddStringAttribute(ax::mojom::StringAttribute::kValue, value);
```

Then down here you could do:
```objc
NSRange range = [native_root accessibilityRangeForPosition:(NSPoint)point];
EXPECT_EQ(range.location, 0);
EXPECT_EQ(range.length, value.length());
```

Reply:
This is a good idea, although, the test won't work with
```
std::string value = "ä";
```
I believe this is because `ä` is one glyph that is composed of two UTF-16 code units.

Reply:
It can be written as a single codepoint or as a combining character sequence. I can't remember if NSString does canonicalisation by default (my guess is probably not) :/ 

[`[NSString rangeOfComposedCharacterSequence]`](https://developer.apple.com/documentation/foundation/nsstring/1416036-rangeofcomposedcharactersequence?language=objc) can get you the range for a character but we'd still need a way to get at the index corresponding to a glyph.

aside: to be totally unambiguous/clear to the reader, you could write this using a unicode escape: `"\u00e4"` for the single codepoint version or `"\u0061\u0308"` for the decomposed version.

Reply:
@cbracken Following up on [this comment](https://github.com/flutter/engine/pull/35453#discussion_r948519493) we might want to leave the test alone, such that it only tests the crash. Right now, it might not be possible for `accessibilityRangeForPosition` to return anything meaningful.

Reply:
It's not from within this part of the code. 

@chunhtai if we delegated this to a method in FlutterPlatformNodeDelegate do you know if there's a reasonable way at getting at glyph-level text metrics from the semantics tree? Off the top of my head I'm not aware of anything.

Reply:
Not that I am aware. If we have to do this, framework will have to send text metrics to the AXTree. To do that, Skia SKParagraph needs to have additional API to expose these metrics so that framework can gets it. I don't think this is worth it unless this method is used in a meaningful way by the macos

Reply:
@a-wallen did some poking around and returning (0,0) triggers a fallback to accessibilityRangeForLine, which then returns the full string, which fixes the app crash and gives us the correct behaviour.

Reply:
@cbracken and @chunhtai, I'd like to merge this PR with 0aa2f5ff93ffe775ad305b9cf5338493af284ec2. Editing `accessibilityRangeForPosition` to return `NSMakeRange(0,0)` causes a fallback on `accessibilityRangeForLine` produces the appropriate behavior for Flutter for the time being.

Reply:
This sounds good to me for now. BTW, @cbracken If iOS needs it, we may consider implement it for real.